inference-engine/src/transformations/src/transformations/snippets/assign_registers_pass.cpp

```diff
@@ -0,0 +1,248 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/snippets/assign_registers_pass.hpp"
+#include "transformations/snippets/remarks.hpp"
+#include "transformations/snippets/register_info.hpp"
+
+#include <ngraph/opsets/opset1.hpp>
+
+#include "transformations/snippets/load.hpp"
+#include "transformations/snippets/scalar.hpp"
+#include "transformations/snippets/broadcastload.hpp"
+
+#include <iterator>
+
+// Assigning internal `vector` register indexes to Ops (virtual registers)
+// After this pass changing order of variables or datafrow lead to invalidation of register assignment
+bool ngraph::pass::AssignRegistersPass::run_on_function(std::shared_ptr<Function> f) {
```

Comment:
Please add `RUN_ON_FUNCTION_SCOPE` macro to disable this code if CC is used.

Reply:
@ilyachur, should I still add this macro even if the code is moved to separate lob?